app/monitoringapi.go

```diff
@@ -56,7 +64,16 @@ func wireMonitoringAPI(life *lifecycle.Manager, addr string, localNode *enode.Lo
 		return errors.New("invalid eth2 service")
 	}
 
-	mux.Handle("/readyz", newReadyHandler(tcpNode, eth2Cl, peerIDs))
+	readyErrFunc := startReadyChecker(ctx, tcpNode, eth2Cl, peerIDs, clockwork.NewRealClock())
+	mux.HandleFunc("/readyz", func(w http.ResponseWriter, r *http.Request) {
+		readyErr := readyErrFunc()
+		if readyErr != nil {
```

Comment:
can inline

Reply:
yeah but i avoided this because it needs send response as error string which will result in calling readErrFunc() twice and again asking for mutex.